File: server/BeerWorkshop.Application/Dto/Requests/Basket/RealizationRequestDto.cs
using BeerWorkshop.Application.Enums;

namespace BeerWorkshop.Application.Dto.Requests.Basket;
public record RealizationRequestDto(IEnumerable<RealizationItemRequestDto> Products, IEnumerable<RealizationItemRequestDto> Devices, string Cashier, int? TotalDiscount = null, DiscountCalculatorType DiscountCalculatorType = DiscountCalculatorType.FullDiscount);

File: client/src/redux/dtos/requests/basket/RealizationRequestDto.ts
import { DiscountCalculatorType } from "../../../enums";
import { RealizationItemRequestDto } from "./RealizationItemRequestDto";

export interface RealizationRequestDto {
  products: RealizationItemRequestDto[];
  devices: RealizationItemRequestDto[];
  totalDiscount: number | null;
  discountCalculatorType: DiscountCalculatorType;
}
